File: 5.py
A, B, C, X, Y = map(int, input().split())
max_c = 2 * max(X, Y)
value = 10**9
for c_num in range(0, max_c+1, 2):
    a_num = max(0, X - c_num//2)
    b_num = max(0, Y - c_num//2)
    tmp = a_num * A + b_num * B + c_num * C
    if tmp <= value:
        value = tmp
print(value)

File: 5.cpp
#include <bits/stdc++.h>
using namespace std;


int main() {
    int A, B, C, X, Y;
    cin >> A >> B >> C >> X >> Y;
    int all_num = 2 * max(X, Y);
    int price = INT_MAX;
    for (int C_num = 0; C_num < all_num+1; C_num++) {
        int A_num = max(0, X - C_num/2);
        int B_num = max(0, Y - C_num/2);
        int tmp_price = A*A_num + B*B_num + C*C_num;
        if (price > tmp_price) {
            price = tmp_price;
        }
    }
    cout << price << endl;
}
